gui-backend/src/main/java/com/rln/gui/backend/implementation/GuiBackendApiImplementation.java

```diff
@@ -158,21 +177,24 @@ public void postWallets(@Valid @NotNull WalletDTO walletDTO) {
     // Get all the balances for all addresses in a wallet
     @Override
     public List<Balance> getBalances(Long walletId) {
-        if (!walletId.equals(ONLY_SUPPORTED_WALLET.getId()))
+        if (ONLY_SUPPORTED_WALLET_ID != walletId) {
```

Comment:
This is generally not a good idea because of `Long` cache. E.g.:

```java
@Test
void long_reference_equality() {
  assertTrue(Long.valueOf(127L) == Long.valueOf(127L));
  assertFalse(Long.valueOf(128) == Long.valueOf(128));
}
```

Reply:
It is gone.